src/client/modules/my/dataUpload/dataUpload.ts

```diff
@@ -74,17 +86,30 @@ export default class DataUploader extends LightningElement {
                 headers: {
                     'Content-Type': 'application/json'
                 },
-                body: JSON.stringify([data]) // Remove brackets for single data insertion
+                body: JSON.stringify(dataList)
             }).then((response) => {
                 //check response from server
                 if (response.status == 200) location.reload();
                 //refresh page
                 else if (response.status >= 400) {
-                    // TODO : Make the error message to the user more descriptive, so they don't have to go into log to see
-                    this.setErrorMessages([
-                        'Something went wrong with the data upload. Please try again.'
-                    ]);
-                    console.error('Error inserting game results: ', response);
+                    //If there has been an error
+
+                    if (response.status == 409) {
+                        //If the error was a duplicate ID
+                        this.setErrorMessages([
+                            'Error: there is a duplicate game id present'
```

Comment:
Can we include the message in the response here, so that it's clear to the user what went wrong?